### ISSUE TITLE
Custom Directives

### ISSUE THREAD

**Issue by t1** [eclipse/microprofile-graphql#344]
I would like to be able to add [Custom Directives](http://spec.graphql.org/draft/#sec-Type-System.Directives.Custom-Directives) to the generated schema. It would be convenient to add an `@Directive` annotation to my own annotation, i.e.:

```java
@Directive(on = { FIELD, ARGUMENT })
public @interface Example {
    String value();
}

@Type
class SomeType {
    @Example("out")
    String field(@Example("in") int arg) { return null; }
}
```

...generates a schema:

```
type SomeType {
    field(arg: Int @example(value: "in")): String @example(value: "out")
}
```

**Comment by t1** [comment]
Some possible use-cases for Custom Directives:

1. BeanValidation: `firstName: String @size(max: 100)` - this is already being checked in the server (at least in WildFly)
2. Request Throttling: `location: String @complexity(score: 30)` - this could also be checked in the backend; see #340
3. RBAC: `location: String @visibleFor(role: "HeroLocator")` - this could also be checked in the backend some day

These directives would add a benefit for the client as "structured documentation".

**Comment by phillip-kruger** [comment]
We already do 3 in the backend as well - what we can do is to make these available in the schema as directives, that would be cool...

**Comment by t1** [comment]
yeah, of course... I just forgot the `@RollesAllowed` annotation which already works, of course.

**Comment by KSmigielski** [comment]
Hi, I would like to help with this issue

**Comment by KSmigielski** [comment]
Hi, I would like to help with this issue

**Comment by t1** [comment]
@KSmigielski : that would be great! If you need help, just ask!
